### PR TITLE
Remove rotation_matrix_* functions

### LINKED ISSUE #294: deprecate rotation_matrix_* functions
I think we can deprecate the `transformations.rotation_matrix_*` functions in favor of `WXRotation.as_matrix` to allow more input options and clean up the API a bit

They are mostly used for testing

### PR DESCRIPTION
**Pull request by vhirtham**
## Changes

Removes the following functions from the `transformations` package:
- `rotation_matrix_x`
- `rotation_matrix_y`
- `rotation_matrix_z`

## Related Issues

Closes #294

## Checks

- [x] updated CHANGELOG.md
- [x] updated tests
- [x] ~~updated doc/~~
- [x] update example/tutorial notebooks

### CODE REVIEW

**Comment by CagtayFabry** [comment]
maybe we could combine this with #153?

**Comment by vhirtham** [comment]
> maybe we could combine this with #153?

I ll try it after everything is working again

**Comment by vhirtham** [comment]
Not sure if I feel good with [an error that disappeared by itself](https://github.com/BAMWelDX/weldx/runs/2227803430) (
commit 9a3df0dd4fd411ebecf29d4f7ba76cf7814751a2). Think it wasn't my/our fault, but maybe one of you (@CagtayFabry @marscher ) have a clue what was going on.

**Comment by CagtayFabry** [comment]
> Not sure if I feel good with [an error that disappeared by itself](https://github.com/BAMWelDX/weldx/runs/2227803430) (
> commit [9a3df0d](https://github.com/BAMWelDX/weldx/commit/9a3df0dd4fd411ebecf29d4f7ba76cf7814751a2)). Think it wasn't my/our fault, but maybe one of you (@CagtayFabry @marscher ) have a clue what was going on.

this seems to pop up with multiprocessing on windows some times .. not sure why

File: tutorials/geometry_02_geometry.ipynb [hunk 1]
```diff
@@ -216,7 +216,7 @@
    },
    "outputs": [],
    "source": [
-    "lcs_rot = LocalCoordinateSystem(tf.rotation_matrix_y(Q_(45, \"degree\")))\n",
+    "lcs_rot = LocalCoordinateSystem(tf.WXRotation.from_euler(\"y\",Q_(45, \"degree\")).as_matrix())\n",
```
Comment (CagtayFabry) [inline tutorials/geometry_02_geometry.ipynb:219]: did this work before?
```python
WXRotation.from_euler("x",Q_(45,"degree")).as_euler("xyz",degrees=True)
>>> array([58.31007809,  0.        ,  0.        ])

**Comment by CagtayFabry** [comment]
cref #318

File: tutorials/geometry_02_geometry.ipynb (hunk 1 above)
Comment (vhirtham) [reply tutorials/geometry_02_geometry.ipynb:219]: Honestly, I didn't check the outputs 😰

**Comment by vhirtham** [comment]
> cref #318

waiting for this to be merged first

**Review by marscher — APPROVED** [review]
LGTM!

**Comment by CagtayFabry** [comment]
what do you think about using some shorter lambda functions in the tests to slim down the code a little? @vhirtham 

So here we could do
```python
rmat = lambda seq, angles, degrees=None: WXRotation.from_euler(seq, angles, degrees).as_matrix()

# old
WXRotation.from_euler("x", angle_x).as_matrix()
#new
rmat("x", angle_x)
```

**Review by CagtayFabry — APPROVED** [review]
thanks! :)

**Comment by vhirtham** [comment]
> what do you think about using some shorter lambda functions in the tests to slim down the code a little? @vhirtham
> (or just a regular private function in 2 lines I guess 😉 )
> 
> So here we could do
> 
> ```python
> rmat = lambda seq, angles, degrees=None: WXRotation.from_euler(seq, angles, degrees).as_matrix()
> 
> # old
> WXRotation.from_euler("x", angle_x).as_matrix()
> #new
> rmat("x", angle_x)
> ```

Think we could do this when we clean up the tests at some point (and there is plenty to clean up). Would leave it for now.

**Pull request merged**